### PR TITLE
[SPARK-46754][SQL][AVRO] Fix compression code resolution in avro table definition and write options

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes the case sensitivity of 'compression' in the avro table definition and the write options, in order to make it consistent with other file sources. Also, the current logic for dealing invalid codec names is unreachable.

### Why are the changes needed?

bugfix

### Does this PR introduce _any_ user-facing change?

yes, 'compression'='Xz', 'compression'='XZ' now works as well as 'compression'='xz'
### How was this patch tested?

new tests

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @dongjoon-hyun @cloud-fan, thank you in advance for reviewing this PR if you have some time.

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroCodecSuite.scala [hunk 1]
```diff
@@ -27,19 +30,34 @@ class AvroCodecSuite extends FileSourceCodecSuite {
...
       }
     }
   }
+
+  test("SPARK-46754: invalid compression codec name in avro table definition") {
+    checkError(
+      exception = intercept[SparkIllegalArgumentException](
+        sql(
+          s"""CREATE TABLE avro_t
+             | USING $format OPTIONS('compression'='unsupported')
```
Comment (dongjoon-hyun) [inline connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroCodecSuite.scala:55]: Although this is a copy of the existing test case pattern, shall we remove the space?
```scala
- | USING $format OPTIONS('compression'='unsupported')
+ |USING $format OPTIONS('compression'='unsupported')
```

File: connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala [hunk 2]
```diff
@@ -102,22 +102,25 @@ private[sql] object AvroUtils extends Logging {
...
-      }
-      job.getConfiguration.set(AvroJob.CONF_OUTPUT_CODEC, codec)
+    parsedOptions.compression.toLowerCase(Locale.ROOT) match {
+      case codecName if AvroCompressionCodec.values().exists(c => c.lowerCaseName() == codecName) =>
+        AvroCompressionCodec.fromString(codecName) match {
+          case UNCOMPRESSED =>
+            job.getConfiguration.setBoolean("mapred.output.compress", false)
+          case compressed =>
+            job.getConfiguration.setBoolean("mapred.output.compress", true)
+            job.getConfiguration.set(AvroJob.CONF_OUTPUT_CODEC, codecName)
```
Comment (dongjoon-hyun) [inline connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala:112]: I guess Avro can accept in a case-insensitive way, but can we do like the following to match with the existing code?
```scala
- job.getConfiguration.set(AvroJob.CONF_OUTPUT_CODEC, codecName)
+ job.getConfiguration.set(AvroJob.CONF_OUTPUT_CODEC, compressed.getCodecName())
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I have only one minor comment.

File: connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala (hunk 2 above)
Comment (yaooqinn) [reply connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala:112]: OK

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroCodecSuite.scala (hunk 1 above)
Comment (yaooqinn) [reply connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroCodecSuite.scala]: I have removed any extra spaces and blank lines from both the origin and this one. Thank you for pointing this out

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for updates.

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun and @cloud-fan, merged to master

**Pull request closed without merging**